### PR TITLE
Revamp JWT docs

### PR DESCRIPTION
**Pull request by marcogschmidt**
Refactor the [JWT docs](https://docs.solo.io/gloo/latest/gloo_routing/virtual_services/security/jwt/).

### CODE REVIEW

File: docs/content/gloo_routing/virtual_services/security/jwt/_index.md [hunk 1]
```diff
@@ -14,15 +14,14 @@ application server. Those who are not will be rejected (usually via an HTTP 401
 
 JWTs are useful in various scenarios, such as:
 
-- OpenID Connect's `id_token` is a JWT. The `id_token` is used to identify the End User (*Resource Owner* 
+- The **OpenID Connect** `id_token` is a JWT. The `id_token` is used to identify the End User (*Resource Owner* 
   in OIDC/OAuth terminology) and is usually sent by the client (phone app or web-browser) to
   the cloud back-end (*Resource Server* in OIDC/OAuth terminology)
 - Kubernetes uses JWT as service accounts secrets within Pods. A program running in a Pod can
```
Comment (GrahamGoudeau) [inline docs/content/gloo_routing/virtual_services/security/jwt/_index.md]: should this be "service ACCOUNTS secrets" or "service ACCOUNT secrets"?

File: docs/content/gloo_routing/virtual_services/security/jwt/_index.md [hunk 2]
```diff
@@ -14,15 +14,14 @@ application server. Those who are not will be rejected (usually via an HTTP 401
...
-  use this JWT to authenticate with the Kuberenetes API server with the permissions of the 
+  use this JWT to authenticate with the Kubernetes API server with the permissions of the 
   service account.
 
-## How is a JWT structured
-
+## How is a JWT structured?
```
Comment (GrahamGoudeau) [inline docs/content/gloo_routing/virtual_services/security/jwt/_index.md]: several titles in this doc have inconsistent capitalization- eg, this one has none apart from the first letter, the next has a mix

File: docs/content/gloo_routing/virtual_services/security/jwt/_index.md [hunk 3]
```diff
@@ -34,19 +33,17 @@ metadata on the JWT (like the signing algorithm). The payload carries `claims` t
 (more on that in the next section). And finally the signature part is a cryptographic signature that 
 signs the header and the payload.
 
-## How does a JWT Carry Identity Information
-
+## How does a JWT Carry Identity Information?
 Inside the JWT various *claims* are encoded; claims provide identity information. A few standard claims are:
```
Comment (GrahamGoudeau) [inline docs/content/gloo_routing/virtual_services/security/jwt/_index.md]: are these standard claims documented somewhere? Might be good to have a quick link off to a place that defines these, but no big deal

File: docs/content/gloo_routing/virtual_services/security/jwt/access_control.md [hunk 4]
```diff
@@ -4,84 +4,94 @@ weight: 2
 description: JWT verification and Access Control (without an external auth server)
 ---
 
-## Prerequisites
+{{% notice note %}}
+The JWT feature was introduced with **Gloo Enterprise**, release 0.13.16. If you are using an earlier version, this tutorial will not work.
+{{% /notice %}}
 
-We will use the following utilities
+In this guide, we will show how to use Gloo to verify kubernetes service account JWTs and how to define RBAC policies to
```
Comment (GrahamGoudeau) [inline docs/content/gloo_routing/virtual_services/security/jwt/access_control.md]:
```suggestion
In this guide, we will show how to use Gloo to verify Kubernetes service account JWTs and how to define RBAC policies to 
```

File: docs/content/gloo_routing/virtual_services/security/jwt/access_control.md [hunk 5]
```diff
@@ -4,84 +4,94 @@ weight: 2
...
 
-Start minikube:
-```shell script
-minikube start
-```
-Make sure your kubectl context is configured to use the `default` namespace:
-```shell script
-kubectl config set-context --current --namespace default
-```
+### Deploy sample application
```
Comment (GrahamGoudeau) [inline docs/content/gloo_routing/virtual_services/security/jwt/access_control.md]: inconsistent title capitalization in this file too

File: docs/content/gloo_routing/virtual_services/security/jwt/access_control.md [hunk 6]
```diff
@@ -120,28 +133,20 @@ FYkg7AesknSyCIVMObSaf6ZO3T2jVGrWc0iKfrR3Oo7WpiMH84SdBYXPaS1VdLC1
...
-The public key is given to the kube api-server in the command line arg `--service-account-key-file`.
-You can see it like so: `minikube ssh ps ax ww | grep kube-apiserver`
+If the above command doesn't produce the expected output, it could be that the `/var/lib/minikube/certs/sa.pub` is 
+different on your minikube. The public key is given to the Kubernetes API server in the `--service-account-key-file` 
+command line flag. You can check which value was passed via this flag by running `minikube ssh ps ax ww | grep kube-apiserver`.
 {{% /notice %}}
 
-Configure JWT verification in Gloo's default virtual service:
+### Secure the Virtual Service
+Now let's configure our Virtual Service to verify JWTs in incoming request using this public key:
```
Comment (GrahamGoudeau) [inline docs/content/gloo_routing/virtual_services/security/jwt/access_control.md]:
```suggestion
Now let's configure our Virtual Service to verify JWTs in the incoming request using this public key:
```

File: docs/content/gloo_routing/virtual_services/security/jwt/access_control.md [hunk 7]
```diff
@@ -4,84 +4,94 @@ weight: 2
 description: JWT verification and Access Control (without an external auth server)
```
Comment (GrahamGoudeau) [inline docs/content/gloo_routing/virtual_services/security/jwt/access_control.md]: I also think this document would benefit from a table of contents

File: docs/content/gloo_routing/virtual_services/security/jwt/access_control.md [hunk 8]
```diff
@@ -362,39 +438,127 @@ returns
...
+{{< tab name="kubectl" codelang="yaml">}}
+apiVersion: gloo.solo.io/v1
+kind: Upstream
+metadata:
+  labels:
+    app: jwks-server
+  name: gloo-system-jwks-server-80
+  namespace: gloo-system
+spec:
+  upstreamSpec:
```
Comment (GrahamGoudeau) [inline docs/content/gloo_routing/virtual_services/security/jwt/access_control.md]: ❌ ❌ ❌ ❌ no more upstreamSpec ❌ ❌ ❌ ❌

File: docs/content/gloo_routing/virtual_services/security/jwt/access_control.md [hunk 9]
```diff
@@ -309,17 +382,18 @@ for production workloads. Use appropriate secret management tools to store sensi
 {{% /notice %}}
 
 ### Create the Json Web Key Set (JWKS)
```
Comment (GrahamGoudeau) [inline docs/content/gloo_routing/virtual_services/security/jwt/access_control.md]: in all these files: Should `Json` be stylized like it is here, or like `JSON`?

File: docs/content/gloo_routing/virtual_services/security/jwt/claim_routing.md [hunk 10]
```diff
@@ -4,78 +4,77 @@ weight: 3
...
+Let's deploy a sample application which will simulate a canary deployment. We will use Hashicorp's 
+[http-echo](https://github.com/hashicorp/http-echo) application, which listens for HTTP requests 
+and echoes back a configurable string. We will deploy:
+ 
+1. a pod that responds with the string "primary" to simulate the primary deployment, 
 ```shell
-openssl genrsa 2048 > private-key.pem
-openssl rsa -in private-key.pem -pubout > public-key.pem
+kubectl run --generator=run-pod/v1 --labels stage=primary,app=echoapp primary-pod \
+    --image=hashicorp/http-echo -- -text=primary -listen=:8080
```
Comment (GrahamGoudeau) [inline docs/content/gloo_routing/virtual_services/security/jwt/claim_routing.md]: this indentation is not coming through in the rendered HTML. It just looks like

```
kubectl run --generator...
--image=hashicorp...
```

File: docs/content/gloo_routing/virtual_services/security/jwt/claim_routing.md [hunk 11]
```diff
@@ -4,78 +4,77 @@ weight: 3
...
+the 'stage' key to create subsets for the service:
+
+```yaml
+apiVersion: gloo.solo.io/v1
+kind: Upstream
+metadata:
+  name: echoapp
+  namespace: gloo-system
+spec:
+  upstreamSpec:
```
Comment (GrahamGoudeau) [inline docs/content/gloo_routing/virtual_services/security/jwt/claim_routing.md]: ❌ upstreamSpec  ❌

File: docs/content/gloo_routing/virtual_services/security/jwt/claim_routing.md [hunk 12]
```diff
@@ -172,65 +133,67 @@ spec:
...
-
-get the url for the proxy
-```
-GATEWAY_URL=$(glooctl proxy url)
-```
-curl as a solo.io team member:
+### Testing our configuration
+Send a request as a solo.io team member:
 ```
 curl "$GATEWAY_URL?token=$SOLO_TOKEN"
```
Comment (GrahamGoudeau) [inline docs/content/gloo_routing/virtual_services/security/jwt/claim_routing.md]: this has the token in a query param, but the virtual service spec says that the tokenSource is the `x-token` header. Is that an inconsistency?

File: docs/content/gloo_routing/virtual_services/security/jwt/claim_routing.md [hunk 13]
```diff
@@ -172,65 +133,67 @@ spec:
...
 ```
 The output should be `canary`.
 
-curl as a othercompany.com team member:
+Send a request as a othercompany.com team member:
 ```
 curl "$GATEWAY_URL?token=$OTHER_TOKEN"
```
Comment (GrahamGoudeau) [inline docs/content/gloo_routing/virtual_services/security/jwt/claim_routing.md]: same here

File: docs/content/gloo_routing/virtual_services/security/jwt/_index.md (hunk 1 above)
Comment (marcogschmidt) [reply docs/content/gloo_routing/virtual_services/security/jwt/_index.md]: The latter

File: docs/content/gloo_routing/virtual_services/security/jwt/_index.md (hunk 3 above)
Comment (marcogschmidt) [reply docs/content/gloo_routing/virtual_services/security/jwt/_index.md]: Good idea, will add a link to the RFC.

File: docs/content/gloo_routing/virtual_services/security/jwt/access_control.md (hunk 5 above)
Comment (marcogschmidt) [reply docs/content/gloo_routing/virtual_services/security/jwt/access_control.md]: Will fix. I will leave expanded acronyms capitalized (e.g. `JSON Web Key Set`). Doing the same for Gloo resource types (e.g. `Virtual Service`).

File: docs/content/gloo_routing/virtual_services/security/jwt/access_control.md (hunk 8 above)
Comment (marcogschmidt) [reply docs/content/gloo_routing/virtual_services/security/jwt/access_control.md]: These docs are pre v1.0.0 so we still need to use this format.

File: docs/content/gloo_routing/virtual_services/security/jwt/access_control.md (hunk 9 above)
Comment (marcogschmidt) [reply docs/content/gloo_routing/virtual_services/security/jwt/access_control.md]: JSON

File: docs/content/gloo_routing/virtual_services/security/jwt/claim_routing.md (hunk 11 above)
Comment (marcogschmidt) [reply docs/content/gloo_routing/virtual_services/security/jwt/claim_routing.md]: Same as above

File: docs/content/gloo_routing/virtual_services/security/jwt/claim_routing.md (hunk 12 above)
Comment (marcogschmidt) [reply docs/content/gloo_routing/virtual_services/security/jwt/claim_routing.md]: This is to be read as "take the query param named `token` and add it to a header named `x-token`:

```yaml
tokenSource:       
  headers:         
  - header: x-token
  queryParams:     
  - token          
```

File: docs/content/gloo_routing/virtual_services/security/jwt/claim_routing.md (hunk 13 above)
Comment (marcogschmidt) [reply docs/content/gloo_routing/virtual_services/security/jwt/claim_routing.md]: See above

**Pull request merged**